source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -157,11 +158,49 @@ ThreadAwareLoadBalancerBase::LoadBalancerImpl::chooseHost(LoadBalancerContext* c
     stats_.lb_healthy_panic_.inc();
   }
 
+
+  /*
+    This will be a config parameter under consistent_hashing_lb_config.
+
+    hash_balance_factor: a percentage of load to bound per upstream server.
+    For eg., with a value of 150, no upstream host will get a load more than
+    1.5 times the average load of any other upstream host.
+
+    Default 0 for being disabled.
+
+  */
+  const uint32_t hash_balance_factor = 125;
+
```

Comment:
Just noting that the API changes will have to be implemented before we accept this -- the current behavior needs to be the default.